### PR TITLE
239 axis scale with data

### LINKED ISSUE #239: Axis scaling with data
Currently our axes scale, but are statically initialized. We need to scale each axis based on the data that we have selected to plot.

### PR DESCRIPTION
**Pull request by LongQuanNguyen**
Closes #239

# What was the issue
- Make each axis scale the interval correctly based on the absolute max values, and round off to the next integer.
- Example: columnX's max value is -3.3. Then the recorded max value will be 4, the x-axis will display from -4 to 4.
# What was done and why
- Modify getPoints in the repository to return absolute-max-round-to-next-int values in an array.
- Modify createGraphingDataPoint and createPosition to take in this max values array.
- Modify each SingleAxis, GenerateXYZ, GenerateTicks to take in their correspond max values to display intervals appropriately.

# How it was tested
- Modify old test cases to match new implementation of functions.
- Add test file and more unit test case.
- Visually test with max values of 10, 100 and 1000 and the graph interval display correctly as below:
image
- Try plotting three points: (1, 10, 100), (5, 50, 500), (10, 100, 1000) and they seem to plot correctly:
image

### CODE REVIEW

File: my-webxr-app/src/App.tsx
Comment (matthew-buglass) [inline my-webxr-app/src/App.tsx:18]: Do we want the graph to start with 10, 100, 1000? Or do we want them all to start the same?

File: my-webxr-app/src/components/CreatePointPositions.tsx
Comment (matthew-buglass) [inline my-webxr-app/src/components/CreatePointPositions.tsx:7]: @pre-condition
@post-condition

File: my-webxr-app/src/components/GenerateTicks.tsx
Comment (matthew-buglass) [inline my-webxr-app/src/components/GenerateTicks.tsx:23]: This only creates a singular Tick. So we shouldn't have the name be plural.

File: my-webxr-app/src/components/SingleAxis.tsx
Comment (matthew-buglass) [inline my-webxr-app/src/components/SingleAxis.tsx:2]: Tick again

File: my-webxr-app/src/components/SingleAxis.tsx
Comment (matthew-buglass) [inline my-webxr-app/src/components/SingleAxis.tsx:23]: @post-condition

File: my-webxr-app/src/contexts/AxesSelectionContext.tsx
Comment (matthew-buglass) [inline my-webxr-app/src/contexts/AxesSelectionContext.tsx:5]: This can be deleted

File: my-webxr-app/src/contexts/AxesSelectionContext.tsx
Comment (matthew-buglass) [inline my-webxr-app/src/contexts/AxesSelectionContext.tsx:65]: Replace with Rollbar

File: my-webxr-app/src/contexts/AxesSelectionContext.tsx
Comment (matthew-buglass) [inline my-webxr-app/src/contexts/AxesSelectionContext.tsx:77]: Replace with Rollbar

File: my-webxr-app/src/contexts/AxesSelectionContext.tsx
Comment (matthew-buglass) [inline my-webxr-app/src/contexts/AxesSelectionContext.tsx:89]: Replace with Rollbar

File: my-webxr-app/tests/components/GenerateTicks.test.tsx
Comment (matthew-buglass) [inline my-webxr-app/tests/components/GenerateTicks.test.tsx:3]: Tick

**Review by matthew-buglass — CHANGES_REQUESTED** [review]
There are a few changes that need to be made, but it looks good overall. I want to wait until Aesha's PR is merged since this is based off that one.

**Comment by matthew-buglass** [comment]
The alternative approach is to merge into Aesha's branch

File: my-webxr-app/src/App.tsx
Comment (LongQuanNguyen) [reply my-webxr-app/src/App.tsx:18]: I should get rid of that. They were mean for visual confirmation, the max will be determined by the column data.

**Review by matthew-buglass — APPROVED** [review]

**Pull request merged**
